internal/broadcaster/utxo_creator.go

```diff
@@ -7,97 +7,114 @@ import (
 	"errors"
 	"fmt"
 	"log/slog"
+	"sync"
 	"time"
 
-	sdkTx "github.com/bitcoin-sv/go-sdk/transaction"
-
 	"github.com/bitcoin-sv/arc/internal/metamorph/metamorph_api"
 	"github.com/bitcoin-sv/arc/pkg/keyset"
+	sdkTx "github.com/bitcoin-sv/go-sdk/transaction"
 )
 
 type UTXOCreator struct {
 	Broadcaster
-	keySets []*keyset.KeySet
+	keySet *keyset.KeySet
+	wg     sync.WaitGroup
+}
+type splittingOutput struct {
+	satoshis uint64
+	vout     uint32
 }
 
-func NewUTXOCreator(logger *slog.Logger, client ArcClient, keySets []*keyset.KeySet, utxoClient UtxoClient, isTestnet bool, opts ...func(p *Broadcaster)) (*UTXOCreator, error) {
+func NewUTXOCreator(logger *slog.Logger, client ArcClient, keySet *keyset.KeySet, utxoClient UtxoClient, isTestnet bool, opts ...func(p *Broadcaster)) (*UTXOCreator, error) {
 	b, err := NewBroadcaster(logger, client, utxoClient, isTestnet, opts...)
 	if err != nil {
 		return nil, err
 	}
 
 	creator := &UTXOCreator{
 		Broadcaster: b,
-		keySets:     keySets,
+		keySet:      keySet,
 	}
 
 	return creator, nil
 }
 
-func (b *UTXOCreator) CreateUtxos(requestedOutputs int, requestedSatoshisPerOutput uint64) error {
-	for _, ks := range b.keySets {
-		b.logger.Info("creating utxos", slog.String("address", ks.Address(!b.isTestnet)))
+func (b *UTXOCreator) Wait() {
+	b.wg.Wait()
+}
+func (b *UTXOCreator) Start(requestedOutputs int, requestedSatoshisPerOutput uint64) error {
+	// Increment the WaitGroup counter
+	b.wg.Add(1)
```

Comment:
Adding to the wait group should happen just before starting the go routine on line 71.

Otherwise if there is an error e.g. on line 57, `wg.Done()` will never be called.